src/control/server/mgmt_pool.go

```diff
@@ -605,7 +598,23 @@ func (svc *mgmtSvc) PoolDestroy(ctx context.Context, req *mgmtpb.PoolDestroyReq)
 	}
 
 	// Now on to the rest of the pool destroy, issue drpc.MethodPoolDestroy.
-	svc.log.Debugf("MgmtSvc.PoolDestroy issuing drpc.MethodPoolDestroy, req:%+v\n", req)
+	// Note that we set req.SvcRanks to all ranks in the system, not the PS
+	// replicas, not the up ranks in the pool. Doing such a "blind" destroy
+	// avoids contacting the PS, who may have already been destroyed by a
+	// previous pool destroy attempt or otherwise unavailable at this point.
+	// Moreover, we will also clean up pool resources on ranks that are now
+	// available but have previously been excluded from the pool.
+	gm, err := svc.sysdb.GroupMap()
+	if err != nil {
+		return nil, err
+	}
+	allRanks := make([]uint32, 0, len(gm.RankEntries))
+	for i := range gm.RankEntries {
+		allRanks = append(allRanks, i.Uint32())
+	}
+	sort.Slice(allRanks, func(i, j int) bool { return allRanks[i] < allRanks[j] })
+	req.SvcRanks = allRanks
```

Comment:
@mjmac, when fighting with the unit tests, I noticed that mgmtSvc.PoolCreate uses

    	allRanks, err := svc.sysdb.MemberRanks(system.AvailableMemberFilter)

and wondered if we might want to use that here with a NonExcludedMemberFilter or something? I'm fine with what we have right now; just want to bring a potentially nicer way to our attention.

Reply:
Oh, hmm. Yes, that would have been a better suggestion on my part. I was looking over the `raft.Database` API for the functionality you needed and missed that. It's probably time to reorganize that code a bit to make it easier to browse, as it's grown organically over time and isn't well-organized.

Reply:
@liw: I was just looking at this again to add as a small follow-up PR and I'm wondering why we wouldn't want to use `AvailableMemberFilter`. If a rank is not Joined, would we want to include it in the broadcast?

If we really want to create a new `NotExcludedMemberFilter` that is the inverse of `ExcludedMemberFilter`, then I assume we would probably want to use it in the `PoolCreate` case too, right?

Reply:
@mjmac, good questions... I think `PoolDestroy` shall broadcast to the same set of ranks as what `GroupMap` would return given the same DB. I agree that this set should not include ranks that are not `Joined`. (As a caveat, I'm not 100% sure what each state means. For instance, does Ready mean that the rank has not yet completed the `Join` gRPC?) Strictly speaking, changing the filter for GroupMap has an upgrade/downgrade impact though.

As to `PoolCreate`, I think we shall broadcast to all or a subset of (if we are to support filtering by other characteristics) what `GroupMap` would return given the same DB.

What do you think?